src/cdf.rs

```diff
@@ -4884,720 +4884,174 @@ static av1_default_coef_cdf: [CdfCoefContext; 4] = [
 ];
 
 pub(crate) unsafe fn rav1d_cdf_thread_update(
-    hdr: *const Rav1dFrameHeader,
+    hdr: &Rav1dFrameHeader,
     dst: *mut CdfContext,
```

Comment:
this raw pointer comes from a data field ultimately

    #[derive(Clone, Copy)]
    #[repr(C)]
    pub union CdfThreadContext_data {
        pub cdf: *mut CdfContext,
        pub qcat: c_uint,
    }

what is the policy on turning that into a mutable reference? It seems to happen quite freely in the codebase.

Reply:
We've been preemptively turning non-null pointers into references where we can and are pretty sure they will be references once we've removed all raw pointers.